src/swap/defi/fantomSonicBridge.ts

```diff
@@ -0,0 +1,171 @@
+import { add, gt, lt, sub } from 'biggystring'
+import {
+  EdgeCorePluginOptions,
+  EdgeSpendInfo,
+  EdgeSwapInfo,
+  EdgeSwapPlugin,
+  EdgeSwapQuote,
+  EdgeSwapRequest,
+  SwapAboveLimitError,
+  SwapBelowLimitError,
+  SwapCurrencyError
+} from 'edge-core-js/types'
+import { BigNumber, ethers, PopulatedTransaction } from 'ethers'
+
+import {
+  getMaxSwappable,
+  makeSwapPluginQuote,
+  SwapOrder
+} from '../../util/swapHelpers'
+import {
+  asyncWaterfall,
+  convertRequest,
+  getAddress,
+  shuffleArray
+} from '../../util/utils'
+import { EdgeSwapRequestPlugin } from '../types'
+import ABI from './abi/FANTOM_SONIC_BRIDGE'
+
+const pluginId = 'fantomsonicbridge'
+
+const swapInfo: EdgeSwapInfo = {
+  pluginId,
+  displayName: 'Fantom/Sonic Bridge',
+  orderUri: undefined,
+  supportEmail: 'support@edge.com'
+}
+
+export function makeFantomSonicBridgePlugin(
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const rpcs = [
+    'https://rpcapi.fantom.network',
+    'https://rpc3.fantom.network',
+    'https://rpc.fantom.network',
+    'https://rpc2.fantom.network',
+    'https://1rpc.io/ftm',
+    'https://fantom-rpc.publicnode.com'
+  ]
+  const bridgeContractAddress = '0x3561607590e28e0848ba3b67074c676d6d1c9953'
+
+  const fetchSwapQuoteInner = async (
+    request: EdgeSwapRequestPlugin
+  ): Promise<SwapOrder> => {
+    const fromAddress = await getAddress(request.fromWallet)
+    const toAddress = await getAddress(request.toWallet)
+
+    if (fromAddress !== toAddress) {
+      throw new Error('From and to addresses must be the same')
+    }
+
+    const providers = rpcs.map(rpc => new ethers.providers.JsonRpcProvider(rpc))
+
+    const call = async (method: string, args: string[] = []): Promise<any> => {
+      return await asyncWaterfall(
```

Comment:
Wrap this with try/catch to add a more descriptive error message about inputs and outputs